File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        Stack<String> stk = new Stack<>();
        String part="";
        for(int i=0; i<s.length(); i++){
            part = String.valueOf(s.charAt(i));
            if(
                !stk.isEmpty() && 
                stk.peek().equals(part)
              ){
                stk.pop();
                continue;
            }
            stk.push(part);
        }
        
        return (stk.isEmpty())?1:0;
    }
}

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    s = list(s)
    stack = []
    idx = 0
    while idx < len(s):
        if not stack:
            stack.append(s[idx])
        elif s[idx] == stack[-1]:
            stack.pop()
        else:
            stack.append(s[idx])
        idx+=1;
    return 0 if stack else 1
        